### ISSUE TITLE
is there a way to specify outputting only certain layer(s)?

### ISSUE THREAD

**Issue by dredenba** [wader/inkmake#1]
I'd like to be able to hide/show layers for the output. Is there a way to specify which layers to include in the output image? Or a way to hide/show layers by name in the command?

Currently I just go into the file manually hide/show each layer as I want, and then run inkmake.

**Comment by wader** [comment]
Hello, not at the moment im afraid. Took a quick look at the command line argument to Inkscape and the only argument that might be able to hide all but one layer is `--export-id=ID` combined with `--export-id-only`. Can you try that out manually? running `inkmake -v` will show you the arguments used to Inkscape if that helps.

One solution to be able to hide/show layers arbitrary would be if Inkmake created modified intermediary SVG files before calling Inkscape. Not sure how tricky that is to do.

**Comment by dredenba** [comment]
okay looks like `--export-id-only` (or its equivalent `-j`) would do it for a single id if that id is the layer/group you want. I also need to include the `-C` argument (aka `--export-area-page` ) to keep the export area as the whole page and not the dimensions of the element for that id.

so something like this works:
`inkscape myFileWithLayers.svg -jC -i layerIdHere --export-png=layerImage.png`

The trickiness i found was knowing what the id of the layer is because its name does not match the name that appears in the "Layers Dialog", but if I crack open the .svg file in a text editor, I can see the id associated with each element by finding the top level "g" element with the `inkscape:groupmode="layer"` and `inkscape:label="<label i see in inkscape layers dialog>"`
i.e.

```
...
<g
     inkscape:label="my top layer"
     inkscape:groupmode="layer"
     id="layer1"
...
```

**Comment by wader** [comment]
Ok! so what is needed it a `page` option to use `--export-area-page` and some way of turn on/off layers.

At first glance it looks like it should't be that hard to generate a modified SVG. Lookup groups id:s (or by label if the id is something internal?) that has attribute `inkscape:groupmode="layer"` and then appropriately add `display="none"` attributes etc.

**Comment by dredenba** [comment]
Yep that sounds about right.

I added a functional start in pull request #2 to export a single layer.

If you did the added step of generating a temporary modified copy of the svg as you describe, then I think it would be able to do multiple layers in one pass. 

The labels on the layer are more useful because thats what you see in Inkscape UI, to find the ids you have to dig a bit which isn't terrible just a bit of a nuisance especially since I didn't find a way to change the layer's id with Inkscape (short of hacking the svg file itself).

**Comment by wader** [comment]
Oh nice, thanks! hope i can take a look at the during the weekend. Yeah using label names is probably the way to go or support both somehow. Would not surprise me if there are editors or SVG generators where id:s are more well known.

**Comment by wader** [comment]
Can you give the change a spin say what you think? also I don't think a `page` option is necessary now when using a modified temp file as we don't change the default export area.

**Comment by dredenba** [comment]
Code changes seems generally very good. The temp file approach allows for a lot more flexibility.
I did run into an issue/bug where the display attribute that is dynamically added or removed is not being respected. In my inkscape file, the layer in the temp file has something like:

```
...
<g display='none' inkscape:groupmode='layer' id='g12586' inkscape:label='homeUnselected' style='display:none' transform='translate(0,-29)'>
... 
```

because the style attribute has a value already specified i think it overrides your dynamically added display attribute. my solution would be either to strip out the display in the style too or update the style attribute.

**Comment by dredenba** [comment]
made a pull request with fix in pull request #3

**Comment by wader** [comment]
Oh nice catch!
